### PR TITLE
 ZEN-27877 Improving loading time of Infrastructure page

### PR DESCRIPTION
**Pull request by karolzen**
https://jira.zenoss.com/browse/ZEN-27879

### CODE REVIEW

File: Products/Zuul/facades/__init__.py [hunk 1]
```diff
@@ -206,8 +204,8 @@ def getObjectBrains(self, uid=None, start=0, limit=50, sort='name',
 
 
     def getDevices(self, uid=None, start=0, limit=50, sort='name', dir='ASC',
-                   params=None, hashcheck=None):
-
+                   params={}, hashcheck=None):
+        statuses = params.pop('status', None)
```
Comment (jpeacock-zenoss) [inline Products/Zuul/facades/__init__.py]: Let's comment this line as its placement here is important to the performance of this method.  I suggest the following:
```
    # Passing a key that's not an index in the catalog will
    # cause a serious performance penalty.  Remove 'status' from
    # the 'params' parameter before passing it on.
    statuses = params.pop('status', None)
```

Comment (karolzen) [inline Products/Zuul/facades/__init__.py]: Good point! :+1:

File: Products/Zuul/facades/__init__.py [hunk 2]
```diff
@@ -207,7 +205,11 @@ def getObjectBrains(self, uid=None, start=0, limit=50, sort='name',
 
     def getDevices(self, uid=None, start=0, limit=50, sort='name', dir='ASC',
                    params=None, hashcheck=None):
-
+        # Passing a key that's not an index in the catalog will
+        # cause a serious performance penalty.  Remove 'status' from
+        # the 'params' parameter before passing it on.
+        params = params if params else {}
```
Comment (jpeacock-zenoss) [inline Products/Zuul/facades/__init__.py]: This line should go before the comment.

File: Products/Zuul/facades/__init__.py [hunk 3]
```diff
@@ -172,8 +172,6 @@ def getObjectBrains(self, uid=None, start=0, limit=50, sort='name',
         query = None
         globFilters = {}
         prodStates = None
-        if params is None:
-            params = {}
```
Comment (jpeacock-zenoss) [inline Products/Zuul/facades/__init__.py]: This should also be restored (or replaced with `params = params if params else {}`.  `getObjectBrains` can be called from other places as well.

Comment (karolzen) [inline Products/Zuul/facades/__init__.py]: Done

File: Products/Zuul/facades/__init__.py (hunk 2 above)
Comment (karolzen) [inline Products/Zuul/facades/__init__.py]: Done

**Pull request merged**